PlayerSkeleton.java

```diff
@@ -2,13 +2,28 @@
 public class PlayerSkeleton {
 
 	//implement this function to have a working system
-	public int pickMove(State s, int[][] legalMoves) {
+	public int pickMove(Game s, int[][] legalMoves) {
+		Strategy strategy = new Strategy();
+		int move = 0;
+		double maxScore = -1;
+		Game simulation;
 		
-		return 0;
+		for(int i = 0; i < legalMoves.length; i++) {
+			simulation = (Game) s.clone();
+			simulation.makeMove(i);
+			double score = simulation.getRowsCleared() + strategy.calculate(simulation);
+			
+			if(score > maxScore) {
+				maxScore = score;
+				move = i;
+			}
+		}
+		
+		return move;
 	}
 	
 	public static void main(String[] args) {
-		State s = new State();
+		Game s = new Game();
```

Comment:
Have you considered of using subtyping, i.e. "State s = new Game();"? I wonder which one is of a better implementation?
Tag: @digawp @jials @nathanajah